atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -449,6 +453,45 @@ def _get_replacement_pullspecs_OSBS_resolution(self, pullspecs):
 
         return replacements
 
+    def _replace_other_metadata_from_csv_modifications(self, operator_manifest):
+        """
+        WILL add/append/update/replace metadata in the operator manifest
+        passed in
+
+        :param operator_manifest: an OperatorManifest, as created by `from_directory()`
+        :type operator_manifest: instance of OperatorManifest
+        :return: None
+        """
+
+        def do_action(operator_manifest, action):
+            mods = self._fetch_operator_csv_modifications()
+            if (mods.get(action)):
+                # Loop over mod data and apply action
+                for action_field in mods[action]:
+                    for item in mods[action][action_field]:
+                        if action == 'append':
+                            if isinstance(mods[action][action_field][item], list):
+                                # Append to existing list
+                                if operator_manifest.csv.data[action_field].get(item):
+                                    for value in mods[action][action_field][item]:
+                                        print(value)
+                                        operator_manifest.csv.data[action_field][item].append(value)
+                                else:  # Add list - it doesn't exist in target CSV
+                                    operator_manifest.csv.data[action_field][item] = mods[action][action_field][item]
+                            else:
+                                #  Raise something - we're expecting a list
+                                pass
+                        elif action == 'update':
+                            # Update existing dict
+                            if operator_manifest.csv.data.get(action_field):
+                                operator_manifest.csv.data[action_field][item] = mods[action][action_field][item]
+                            else:  # Add dict ("object") - it doesn't exist in target CSV
+                                operator_manifest.csv.data[action_field]= mods[action][action_field]
+
+        if self._fetch_operator_csv_modifications():
```

Comment:
This is fetched 3 times, should be just once IMHO

Reply:
@MartinBasti; Agreed. Should I make it a field/property of `PinOperatorDigestsPlugin`?

Reply:
just call it once at the begin of _replace_other_metadata_from_csv_modifications and then use it

Reply:
@rcerven; That leaves us with a total of two calls

Reply:
huh? why 2? 1 call is on orchestrator another on worker